CMakeModules/FindFFTW.cmake

```diff
@@ -20,100 +20,76 @@
-    SET(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_STATIC_LIBRARY_SUFFIX})
-ELSE()
-    SET(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_SHARED_LIBRARY_SUFFIX})
-ENDIF()
-
-IF ("${SIZE_OF_VOIDP}" EQUAL 8)
-  SET(MKL_LIB_DIR_SUFFIX "intel64")
-ELSE()
-  SET(MKL_LIB_DIR_SUFFIX "ia32")
-ENDIF()
-
-IF(FFTW_ROOT)
-    #find libs
-    FIND_LIBRARY(
-        FFTW_LIB
-        NAMES "fftw3" "libfftw3-3" "fftw3-3" "mkl_rt"
-        PATHS ${FFTW_ROOT}
-        PATH_SUFFIXES "lib" "lib64" "lib/${MKL_LIB_DIR_SUFFIX}"
-        NO_DEFAULT_PATH
-        )
-    FIND_LIBRARY(
-        FFTWF_LIB
-        NAMES "fftw3f" "libfftw3f-3" "fftw3f-3" "mkl_rt"
-        PATHS ${FFTW_ROOT}
-        PATH_SUFFIXES "lib" "lib64" "lib/${MKL_LIB_DIR_SUFFIX}"
-        NO_DEFAULT_PATH
-        )
-
-    #find includes
-    FIND_PATH(
-        FFTW_INCLUDES
-        NAMES "fftw3.h"
-        PATHS ${FFTW_ROOT}
-        PATH_SUFFIXES "include" "include/fftw"
-        NO_DEFAULT_PATH
-        )
-ELSE()
-    FIND_LIBRARY(
-        FFTW_LIB
-        NAMES "fftw3" "mkl_rt"
-        PATHS ${PKG_FFTW_LIBRARY_DIRS} ${LIB_INSTALL_DIR}
-        )
-    FIND_LIBRARY(
-        FFTWF_LIB
-        NAMES "fftw3f" "mkl_rt"
-        PATHS ${PKG_FFTW_LIBRARY_DIRS} ${LIB_INSTALL_DIR}
-        )
-    FIND_PATH(
-        FFTW_INCLUDES
-        NAMES "fftw3.h"
-        PATHS ${PKG_FFTW_INCLUDE_DIRS} ${INCLUDE_INSTALL_DIR}
-        )
-ENDIF(FFTW_ROOT)
-
-SET(FFTW_LIBRARIES ${FFTW_LIB} ${FFTWF_LIB})
-
-SET(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES_SAV})
-
-INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(FFTW DEFAULT_MSG
-    FFTW_INCLUDES FFTW_LIBRARIES)
+find_package(PkgConfig)
+pkg_check_modules(PKG_FFTW "fftw3")
+
+find_path( FFTW_INCLUDE_DIR
+  NAMES "fftw3.h"
+  PATHS ${FFTW_ROOT}
+        ${CMAKE_SYSTEM_INCLUDE_PATH}
+        ${CMAKE_SYSTEM_PREFIX_PATH}
+        ${PKG_FFTW_INCLUDE_DIRS}
+  PATH_SUFFIXES "include" "include/fftw"
+  )
+
+find_library( FFTW_LIBRARY
+  NAMES "fftw3" "libfftw3-3" "fftw3-3" "mkl_rt"
+  PATHS ${FFTW_ROOT}
+        ${CMAKE_SYSTEM_PREFIX_PATH}
+        ${PKG_FFTW_LIBRARY_DIRS}
+  PATH_SUFFIXES "lib" "lib64" "lib/intel64" "lib/ia32"
+)
+
+find_library( FFTWF_LIBRARY
+  NAMES "fftw3f" "libfftw3f-3" "fftw3f-3" "mkl_rt"
+  PATHS ${FFTW_ROOT}
+        ${CMAKE_SYSTEM_PREFIX_PATH}
+        ${CMAKE_SYSTEM_LIBRARY_PATH}
+        ${PKG_FFTW_LIBRARY_DIRS}
+  PATH_SUFFIXES "lib" "lib64" "lib/intel64" "lib/ia32"
+)
+
+mark_as_advanced(FFTW_INCLUDE_DIR FFTW_LIBRARY FFTWF_LIBRARY)
+
+# if(WIN32)
+#   set(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
+#   get_filename_component(FFTW_LIBRARY_DIR ${FFTW_LIBRARY} DIRECTORY)
+#   find_library( FFTW_LIBRARY_DLL
+#     NAMES "fftw3" "libfftw3-3" "fftw3-3" "mkl_rt"
+#     PATHS ${FFTW_ROOT}
+#           ${CMAKE_SYSTEM_PREFIX_PATH}
+#           ${PKG_FFTW_LIBRARY_DIRS}
+#           ${FFTW_LIBRARY_DIR}
+#     PATH_SUFFIXES "lib" "lib64" "lib/intel64" "lib/ia32"
+#   )
+#
+#   find_library( FFTWF_LIBRARY_DLL
```

Comment:
If these commented out sections are not being used anymore, can you please remove them.

Reply:
Addressed